packages/roosterjs-editor-core/lib/coreApi/getPendableFormatState.ts

```diff
@@ -18,16 +18,15 @@ export const getPendableFormatState: GetPendableFormatState = (
     forceGetStateFromDOM: boolean
 ): PendableFormatState => {
     const range = core.api.getSelectionRange(core, true /* tryGetFromCache*/);
-    const isRangeCollapsed = range && range.collapsed;
     const cachedPendableFormatState = core.pendingFormatState.pendableFormatState;
     const cachedPosition = core.pendingFormatState.pendableFormatPosition?.normalize();
-    const currentPosition = isRangeCollapsed && Position.getStart(range).normalize();
+    const currentPosition = range && Position.getStart(range).normalize();
     const isSamePosition = currentPosition && currentPosition.equalTo(cachedPosition);
```

Comment:
Now currentPosition will have value even range is not collapsed. So you also need to check range is collapsed here